src/foam/doc/DocBrowser.js

```diff
@@ -801,6 +807,10 @@ foam.CLASS({
       var nbrOfPropInNonConventionalDiag = 5;
       var propertyHeight = 20;
       this.className  = this.data.name;
+
+      if ( Array.isArray(this.allowedModels) && this.allowedModels.length && this.allowedModels.includes(this.className))
```

Comment:
Would be faster if these were maps instead of arrays. The adapt methods on the properties could convert from arrays to maps internally.